glean-core/android/src/main/java/mozilla/telemetry/glean/private/TimingDistributionMetricType.kt

```diff
@@ -74,3 +88,32 @@ class TimingDistributionMetricType(meta: CommonMetricData, timeUnit: TimeUnit) :
     @VisibleForTesting(otherwise = VisibleForTesting.NONE)
     fun testGetNumRecordedErrors(error: ErrorType) = inner.testGetNumRecordedErrors(error)
 }
+
+class LocalTimingDistribution(
+    val metric: TimingDistributionMetricType,
+    val inner: FfiLocalTimingDistribution,
+) : AutoCloseable {
+    /**
+     * Accumulates one sample into the histogram.
+     *
+     * The provided sample must be in the "unit" declared by the instance of the metric type
+     * (e.g. if the instance this method was called on is using [`crate::TimeUnit::Second`], then
+     * `sample` is assumed to be in seconds).
+     *
+     * Accumulation happens in-memory only.
+     */
+    fun accumulate(sample: Long) = inner.accumulate(sample.toULong())
+
+    /**
+     * Record data from the histogram buffer into the associated metric.
+     *
+     * This instance must not be used after `commit()` is called.
```

Comment:
Is there a way to prevent it from being used after `commit()`? I think the Java-like pattern is to`throw new IllegalStateException("Use after commit");`

Reply:
Hmmm.
Then we need to track that state.

I'm considering to remove the Kotlin part. The first user is Rust only.